pkg/framework/aries/api/config.go

```diff
@@ -0,0 +1,21 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package api
+
+//Config contains configuration items for aries.
+type Config interface {
+	AgentLabel() string
+	AgentServiceEndpoint() string
+}
+
+//ConfigProvider provides config backend for aries
+type ConfigProvider func() (ConfigBackend, error)
+
+//ConfigBackend backend for all config types in aries
+type ConfigBackend interface {
```

Comment:
I am confused by all this indirection around configuration. Why two interfaces? Why interfaces at all?